exporter/src/test/scala/uk/gov/nationalarchives/consignmentexport/BagMetadataSpec.scala

```diff
@@ -0,0 +1,93 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.time.ZonedDateTime
+import java.util.UUID
+
+import cats.effect.IO
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment
+import graphql.codegen.GetConsignmentExport.getConsignmentForExport.GetConsignment.{Series, TransferringBody}
+import org.keycloak.representations.idm.UserRepresentation
+import uk.gov.nationalarchives.consignmentexport.Config.{Api, Auth, Configuration, EFS, S3}
+
+class BagMetadataSpec extends ExportSpec {
+
+  private val fixedDateTime = ZonedDateTime.now()
+  private val userId = UUID.randomUUID()
+  private val series = Series(Some("series-code"))
+  private val transferringBody = TransferringBody(Some("tb-code"))
+  private val consignmentId = UUID.fromString("50df01e6-2e5e-4269-97e7-531a755b417d")
+  private val consignment = GetConsignment(
+    userId, Some(fixedDateTime), Some(fixedDateTime), Some(fixedDateTime), Some(series), Some(transferringBody)
+  )
+  private val config = Configuration(S3("", "", ""), Api(""), Auth("authUrl", "clientId", "clientSecret", "realm"), EFS(""))
+  private val userRepresentation = new UserRepresentation()
+  userRepresentation.setFirstName("FirstName")
+  userRepresentation.setLastName("LastName")
+
+  "the getBagMetadata method" should "return the correct bag metadata for the given consignment id" in {
+
+    val mockKeycloakClient = mock[KeycloakClient]
+    val mockGraphQlApi = mock[GraphQlApi]
+
+    doAnswer(() => IO.pure(Some(consignment))).when(mockGraphQlApi).getConsignmentMetadata(config, consignmentId)
+    doAnswer(() => userRepresentation).when(mockKeycloakClient).getUserDetails(any[String])
+
+    val bagMetadata = BagMetadata(mockGraphQlApi, mockKeycloakClient).getBagMetadata(consignmentId, config).unsafeRunSync()
+    bagMetadata.get("Consignment-Series").get(0) should be("series-code")
+    bagMetadata.get("Source-Organization").get(0) should be("tb-code")
+    bagMetadata.get("Consignment-StartDate").get(0) should be(fixedDateTime.toString)
+    bagMetadata.get("Consignment-CompletedDate").get(0) should be(fixedDateTime.toString)
+    bagMetadata.get("Contact-Name").get(0) should be("FirstName LastName")
+    bagMetadata.get("Consignment-ExportDate").get(0) should be(fixedDateTime.toString)
+  }
+
+  "the getBagMetadata method" should "return the correct bag metadata if a consignment metadata property is missing" in {
```

Comment:
What does "correct" mean in this context? Does it mean that it just returns the metadata that is available?